src/nvim/tui/tui.c

```diff
@@ -709,6 +729,22 @@ static void update_attrs(TUIData *tui, int attr_id)
     }
   }
 
+  if ((attrs.url.size != tui->url.size) || (!strnequal(attrs.url.data, tui->url.data,
+                                                       tui->url.size))) {
+    if (tui->url.data != NULL) {
+      out(tui, S_LEN("\x1b]8;;\x1b\\"));
+    }
+
+    if (attrs.url.data != NULL) {
+      StringBuilder sb = KV_INITIAL_VALUE;
+      kv_printf(sb, "\x1b]8;;%s\x1b\\", attrs.url.data);
```

Comment:
Should include an `id` here. `attr_id` maybe?

```suggestion
      kv_printf(sb, "\x1b]8;id=%d;%s\x1b\\", attr_id, attrs.url.data);
```

Reply:
This is tricky. `attr_id` will make the terminal highlight all URLs with the same attr_id, which we probably don't want. We don't really have a way to uniquely identify each individual hyperlink right now. We could use something like grid position where the URL starts, but don't have a way to easily get that information at the moment.

For now I'll omit the id, and we can solve this if/when it becomes a problem.